lib/taskana-core-test/src/test/java/acceptance/task/requestchanges/RequestChangesWithSpiAccTest.java

```diff
@@ -0,0 +1,267 @@
+      this.taskanaEngine = taskanaEngine;
+    }
+
+    @Override
+    public Task afterRequestChanges(Task task) throws Exception {
+      task.setCustomField(TaskCustomField.CUSTOM_3, "bla");
+      task = taskanaEngine.getTaskService().updateTask(task);
+      task = taskanaEngine.getTaskService().transfer(task.getId(), NEW_WORKBASKET_KEY, "DOMAIN_A");
+      return task;
+    }
+  }
+
+  static class ClassificationUpdater implements AfterRequestChangesProvider {
+
+    public static final String SPI_CLASSIFICATION_NAME = "Neuer Classification Name";
+
+    private TaskanaEngine taskanaEngine;
+
+    @Override
+    public void initialize(TaskanaEngine taskanaEngine) {
+      this.taskanaEngine = taskanaEngine;
+    }
+
+    @Override
+    public Task afterRequestChanges(Task task) throws Exception {
+      Classification newClassification =
+          defaultTestClassification().buildAndStore(taskanaEngine.getClassificationService());
+
+      task.setClassificationKey(newClassification.getKey());
+      task = taskanaEngine.getTaskService().updateTask(task);
+
+      newClassification.setName(SPI_CLASSIFICATION_NAME);
+      taskanaEngine.getClassificationService().updateClassification(newClassification);
+
+      return task;
+    }
+  }
+
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  @WithServiceProvider(
+      serviceProviderInterface = AfterRequestChangesProvider.class,
+      serviceProviders = TaskModifierAndTransferrer.class)
+  class SpiModifiesTask {
+
+    @TaskanaInject TaskService taskService;
+
+    @WithAccessId(user = "user-1-1")
+    @Test
+    void should_ReturnModifiedTask_When_SpiModifiesAndTransfersTask() throws Exception {
+      Task task = createTaskInReviewByUser("user-1-1").buildAndStore(taskService);
+
+      Task result = taskService.requestChanges(task.getId());
+
+      assertThat(result.getCustomField(TaskCustomField.CUSTOM_3)).isEqualTo("bla");
+      assertThat(result.getWorkbasketSummary()).isEqualTo(newWorkbasket);
+    }
+
+    @WithAccessId(user = "user-1-1")
+    @Test
+    void should_ReturnPersistentModifiedTask_When_SpiModifiesAndTransfersTask() throws Exception {
+      Task task = createTaskInReviewByUser("user-1-1").buildAndStore(taskService);
+
+      taskService.requestChanges(task.getId());
+      Task result = taskService.getTask(task.getId());
+
+      assertThat(result.getCustomField(TaskCustomField.CUSTOM_3)).isEqualTo("bla");
+      assertThat(result.getWorkbasketSummary()).isEqualTo(newWorkbasket);
+    }
+  }
+
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  @WithServiceProvider(
+      serviceProviderInterface = AfterRequestChangesProvider.class,
+      serviceProviders = ClassificationUpdater.class)
+  class SpiModifiesTaskAndClassification {
+
+    @TaskanaInject TaskService taskService;
+
+    @WithAccessId(user = "user-1-1", groups = "businessadmin")
+    @Test
+    void should_ReturnUpdateMultipleEntities_When_SpiModifiesTask() throws Exception {
```

Comment:
this name makes no sense to me :sweat_smile: .
Same applies to the other test class.

Reply:
Suggestion: `should_PersistDifferentEntityChanges_When_SpiModifiesMoreThanTheTask` or something like that?

Reply:
What do you think of `should_ChangeMultipleEntities_When_SpiModifiesMoreThanTheTask`? It sounds a little bit simpler to me.

Reply:
sure